airflow/www/static/js/dag/details/taskInstance/TaskFailedDependency.tsx

```diff
@@ -0,0 +1,70 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from "react";
+import { Text, Table, Tbody, Tr, Th, Td, Box } from "@chakra-ui/react";
+
+import { useTaskFailedDependency } from "src/api";
+
+interface Props {
+  dagId: string;
+  runId: string;
+  taskId: string;
+  mapIndex?: number;
+}
+
+const TaskFailedDependency = ({ dagId, runId, taskId, mapIndex }: Props) => {
+  const { data: dependencies, isLoading } = useTaskFailedDependency({
+    dagId,
+    taskId,
+    runId,
+    mapIndex,
+  });
+
+  if (isLoading) {
+    return null;
+  }
+
+  return (
+    <Box mt={3} flexGrow={1}>
+      <Text as="strong" mb={3}>
+        Task Failed Dependencies
+      </Text>
+      <br />
+      <br />
+      <Text>Dependencies Blocking Task From Getting Scheduled</Text>
+      <Table variant="striped">
+        <Tbody>
+          <Tr>
+            <Th>Dependency</Th>
+            <Th>Reason</Th>
+          </Tr>
+          {dependencies?.dependencies?.map((dep) => (
```

Comment:
We could also put a spinner or skeleton here during loading. But we may want to return null if there are no dependencies at all.